Source/WebCore/platform/graphics/angle/GraphicsContextGLANGLE.cpp

```diff
@@ -2938,15 +2940,22 @@ void GraphicsContextGLANGLE::getActiveUniformBlockiv(PlatformGLObject program, G
 }
 
 #if ENABLE(WEBXR)
-GCGLExternalImage GraphicsContextGLANGLE::createExternalImage(ExternalImageSource&&, GCGLenum, GCGLint)
+void GraphicsContextGLANGLE::bindExternalImage(GCGLenum target, GCGLExternalImage image)
 {
-    notImplemented();
-    return { };
-}
-
-void GraphicsContextGLANGLE::bindExternalImage(GCGLenum, GCGLExternalImage)
-{
-    notImplemented();
+    if (!makeContextCurrent())
+        return;
+    EGLImage eglImage = EGL_NO_IMAGE_KHR;
+    if (image) {
+        eglImage = m_eglImages.get(image);
+        if (!eglImage) {
+            addError(GCGLErrorCode::InvalidOperation);
+            return;
+        }
+    }
+    if (target == RENDERBUFFER)
+        GL_EGLImageTargetRenderbufferStorageOES(RENDERBUFFER, eglImage);
+    else
+        GL_EGLImageTargetTexture2DOES(target, eglImage);
```

Comment:
I think all these WEbXR methods should be implemented in GraphicsContextGLTextureMapperGBM

Reply:
Wouldn't it be more flexible to have them here? It would allow platforms not using GBM to run that. Or is that combination not usually supported?